craft_parts/packages/deb.py

```diff
@@ -304,15 +304,22 @@ def _get_dpkg_list_path(base: str) -> pathlib.Path:
 
 
 def _get_filtered_stage_package_names(
-    *, base: str, package_list: List[DebPackage]
+    *,
+    base: str,
+    package_list: List[DebPackage],
+    base_package_names: Optional[Iterable[str]],
 ) -> Set[str]:
     """Get filtered packages by name only - no version or architectures."""
-    manifest_packages = [p.name for p in get_packages_in_base(base=base)]
-    stage_packages = [p.name for p in package_list]
+    if base_package_names:
+        filtered_packages = set(base_package_names)
```

Comment:
Another thing that just occurred to me: since `cls.stage_packages_filters` is an iterable (the result of `itertools.chain()`, what happens when it is read more than once? Doesn't reading from it (to build the set here) consume the iterator? So what if two parts have stage-packages, so that this function gets called multiple times?

Reply:
You're right, this should be a set -- and in this case it could be better to just pass it as a LifecycleManager parameter since all other parameters are passed there. I'll prepare a new PR with this alternative to see how it would look like.

Reply:
nice, but now this extra `set()` can be removed right?